datalad_registry/tasks/__init__.py

```diff
@@ -299,3 +344,199 @@ def process_dataset_url(dataset_url_id: StrictInt) -> None:
+                    ),
+                ),
+            )
+        )
+        .with_for_update(skip_locked=True)  # Skipping already locked rows
+        .order_by(
+            case(
+                (
+                    requested_not_chked_cond,
+                    1,
+                ),
+                (
+                    requested_chked_cond,
+                    2,
+                ),
+                else_=3,
+            ),
+            case(
+                (
+                    requested_not_chked_cond,
+                    RepoUrl.chk_req_dt,
+                ),
+                (
+                    requested_chked_cond,
+                    RepoUrl.last_chk_dt,
+                ),
+                else_=relevant_action_dt,
+            ),
+        )
+        .limit(max_chks_to_dispatch)
+    ).all()
+
+    for id_, last_chk_dt in result:
+        chk_url_to_update.apply_async(
+            (id_, last_chk_dt), expires=chk_url_task_expiration
+        )
+
+
+@shared_task(rate_limit="10/m")
+@validate_arguments
+def chk_url_to_update(
+    url_id: StrictInt, initial_last_chk_dt: Optional[datetime]
+) -> ChkUrlStatus:
+    """
+    Check a dataset url for potential update
+
+    If an update is available, update the clone of the dataset at the url in the local
+    cache and update the corresponding RepoUrl object in the database.
+
+    :param url_id: The id (primary key) of the dataset url, represented by a `RepoUrl`,
+                   in the database
+    :param initial_last_chk_dt: The value of `last_chk_dt` of the `RepoUrl`
+                                when this check was initiated.
+    :raise: ValueError if the RepoUrl of the specified ID does not exist or has not
+            been processed yet.
+    """
+
+    # Select and lock the RepoUrl identified by the given ID if it is not locked
+    # by another transaction
+    url = (
+        db.session.execute(
+            select(RepoUrl).filter_by(id=url_id).with_for_update(skip_locked=True)
+        )
+        .scalars()
+        .one_or_none()
+    )
+
+    if url is None:
+        # ===
+        # There is no RepoUrl in the database with the specified ID, possibly due to
+        # deletion, or the RepoUrl identified by the given ID is currently locked.
+        # ===
+        return ChkUrlStatus.ABORTED
+
+    # ===
+    # At this point, the RepoUrl identified by the given ID is obtained and exclusively
+    # locked by the current transaction
+    # ===
+
+    _validate_url_is_processed(url)
+
+    if url.last_chk_dt != initial_last_chk_dt:
+        # The RepoUrl has been checked by another process since this check was initiated
+        # Skip this check
+        return ChkUrlStatus.SKIPPED
+
+    # ===
+    # The RepoUrl has not been checked by any other process
+    # since this check was initiated
+    # ===
+
+    # todo: the following is just a template
+    try:
+        # Do the checking and potential updating
+        pass
```

Comment:
```
# if DB information is different from what we have
if url.head != ds.repo.get_hexsha():
    # trigger the metadata and stats re-extraction
```

and may be remove that `"origin/HEAD"` where you now do `get_hexsha` and `.describe`